drivers/net/wireless/ath/ath10k/mac.c

```diff
@@ -7992,6 +8060,10 @@ static int ath10k_mac_init_rd(struct ath10k *ar)
 	int ret;
 	u16 rd;
 
+	ret = ath10k_mac_set_regulatory_from_smbios(ar);
+	if (!ret)
+		return 0;
```

Comment:
Shouldn't we return an error for `!ret`?
Also, let's say that `ath10k_mac_set_regulatory_from_smbios` is successful and `ar->ath_common.regulatory.current_rd` is correctly set, shouldn't we just exit from this function to avoid that the value in `ar->ath_common.regulatory.current_rd` could be overwritten few lines ahead?